pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/ClassNamingConventionsRule.java

```diff
@@ -31,6 +33,9 @@ public class ClassNamingConventionsRule extends AbstractNamingConventionRule<AST
     private final PropertyDescriptor<Pattern> enumerationRegex = defaultProp("enum").build();
     private final PropertyDescriptor<Pattern> annotationRegex = defaultProp("annotation").build();
     private final PropertyDescriptor<Pattern> utilityClassRegex = defaultProp("utility class").build();
+    private final PropertyDescriptor<Pattern> testClassRegex = defaultProp("test class")
+            .desc("Regex which applies to test class names. Since PMD 6.52.0.")
+            .defaultValue("[A-Z][a-zA-Z0-9]*Test").build();
```

Comment:
We probably should change the pattern:

> The class name 'AfterAdviceFirstTests' doesn't match '[A-Z][a-zA-Z0-9]*Test'

```suggestion
            .defaultValue("[A-Z][a-zA-Z0-9]*Tests?").build();
```